src/overlays/actors/ovl_En_Wdhand/z_en_wdhand.c

```diff
@@ -105,63 +174,679 @@ static DamageTable D_80AF64F4 = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Wdhand/EnWdhand_Draw.s")
+void EnWdhand_Init(Actor* thisx, PlayState* play) {
+    EnWdhand* this = THIS;
+    s32 i;
+    Vec3f initVel;
+
+    SkelAnime_InitFlex(play, &this->skelAnime, &gDexihandSkel, &gDexihandIdleAnim, this->jointTable, this->morphTable,
+                       DEXIHAND_LIMB_MAX);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    // Constructs an affine transformation that maps world origin to actor origin and rotation
+    // (model -> world but without scaling)
+    Matrix_Translate(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z, MTXMODE_NEW);
+    Matrix_RotateYS(this->actor.shape.rot.y, MTXMODE_APPLY);
+    Matrix_RotateXS(this->actor.shape.rot.x, MTXMODE_APPLY);
+    Matrix_RotateZS(this->actor.shape.rot.z, MTXMODE_APPLY);
+    // Get a world position offset by 3 units
+    Matrix_MultVecY(3.0f, &this->endPoints[EN_WDHAND_HAND_POINT]);
+    // Save this matrix for later
+    Matrix_MtxFCopy(&this->relativeToWorldTransform, Matrix_GetCurrent());
+    // Write the rotation back to the actor
+    Matrix_MtxFToYXZRot(&this->relativeToWorldTransform, &this->actor.shape.rot, false);
+    // Transpose the matrix, since it's a rotation matrix this creates the inverse rotation matrix
+    Matrix_Transpose(&this->relativeToWorldTransform);
+
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+
+    EnWdhand_GetInitVelocity(this, &initVel);
+    this->actor.speed = initVel.z;
+    this->actor.velocity.y = initVel.y;
+    this->actor.velocity.x = Math_SinS(this->actor.world.rot.y) * this->actor.speed;
+    this->actor.velocity.z = Math_CosS(this->actor.world.rot.y) * this->actor.speed;
+
+    Collider_InitAndSetJntSph(play, &this->collider, &this->actor, &sJntSphInit, this->colliderElements);
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderElements); i++) {
+        ColliderJntSphElement* elem = &this->collider.elements[i];
+
+        elem->dim.worldSphere.radius = elem->dim.modelSphere.radius;
+        EnWdhand_Vec3fToVec3s(&elem->dim.worldSphere.center, &this->actor.world.pos);
+    }
+
+    for (i = 0; i < EN_WDHAND_NUM_SEGMENTS; i++) {
+        this->limbScaleFactors[i] = 1.0f;
+    }
+
+    this->limbIndexBeforeCut = -1;
+    this->limbIndexAfterCut = 0;
+
+    this->actor.world.rot.x = 0;
+    this->actor.world.rot.z =
+        Math_Atan2S_XY(this->relativeToWorldTransform.mf[2][1], this->relativeToWorldTransform.mf[0][1]);
+    this->actor.hintId = TATL_HINT_ID_DEXIHAND;
+    EnWdhand_SetupIdle(this);
+}
+
+void EnWdhand_Destroy(Actor* thisx, PlayState* play) {
+    EnWdhand* this = THIS;
+
+    Collider_DestroyJntSph(play, &this->collider);
+}
+
+void EnWdhand_GetInitVelocity(EnWdhand* this, Vec3f* dst) {
+    s32 param;
+
+    param = EN_WDHAND_GET_Y_INIT_VELOCITY(&this->actor);
+    if (param == 0x7F) {
+        param = 0x28;
```

Comment:
```suggestion
        param = 40;
```
seems better in decimal imo